sketch/hash_min.hpp

```diff
@@ -45,7 +46,7 @@ class MinHash : public HashBase<T> {
 
         for (size_t si = 0; si < this->sketch_dim; si++) {
             T min_char = T(0);
-            size_t min_rank = this->hash_size + 1;
+            auto min_rank = std::numeric_limits<decltype(this->hash(0, 0))>::max();
```

Comment:
I find this very hard to read. Sure, you get an automatic type change with the type of the hash() function, but IMO the ugliness of the code is not worth it. I find the existing code is easier to read.

Reply:
hardcoded `uint32_t` since that's what `hash` returns.